### PR TITLE
Next.js -> Vite + React

### PR DESCRIPTION
**Pull request by voluntas**
This pull request involves a significant migration from Next.js to Vite/React, along with various other updates and improvements. The most important changes include environment variable updates, package and configuration adjustments, and code refactoring to accommodate the migration.

Migration from Next.js to Vite/React:

* Updated environment variable prefixes from `NEXT_PUBLIC_` to `VITE_` in `.env.production` and `.env.template` files. [[1]](diffhunk://#diff-48f73ea2653e45c41f31afc2d42dfde14d8caf26d8dd24d3d2a81a52604f4cb6L1-R5) [[2]](diffhunk://#diff-749e06f64632f62a0c0dfbf4c4f3850e27e94ac109aa121fabd5c29469ae88deL1-R5)
* Removed Next.js-related scripts and dependencies from `package.json` and added Vite-related ones. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L16-R21) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R50-R62)
* Created a new `index.html` file for Vite and updated the `playwright.config.ts` to use Vite for the web server. [[1]](diffhunk://#diff-0eb547304658805aad788d320f10bf1f292797b5e6d745a3bf617584da017051R1-R16) [[2]](diffhunk://#diff-f679bf1e58e8dddfc6cff0fa37c8e755c8d2cfc9e6b5dc5520a5800beba59a92L35-R35)

Code Refactoring:

* Refactored `src/DevTools.tsx` (renamed from `src/pages/devtools.tsx`) to remove Next.js-specific components and updated imports accordingly. [[1]](diffhunk://#diff-ff18c84c9012b7d820be58eafe2b6eeaee3ea20ee5eb41bc6f719f571b9a39feL10) [[2]](diffhunk://#diff-ff18c84c9012b7d820be58eafe2b6eeaee3ea20ee5eb41bc6f719f571b9a39feL27)
* Removed the `Head` component as it was specific to Next.js.
* Updated the `Video` component to handle `null` streams and reset the video element appropriately.

Configuration and Dependency Updates:

* Added new entries to `.gitignore` and updated `biome.jsonc` to include `src/` and `tests/` directories. [[1]](diffhunk://#diff-d975bf659606195d2165918f93e1cf680ef68ea3c9cab994f033705fea8238b2R14-R35) [[2]](diffhunk://#diff-955d6dd0b9089fa951f33ccabbab4597fe019091b7bb81e166df2b34dea4a906R3-R12)
* Updated Node.js version requirement to 22.11 and added new dev dependencies such as `vite` and `@types/react-dom`. [[1]](diffhunk://#diff-d975bf659606195d2165918f93e1cf680ef68ea3c9cab994f033705fea8238b2R14-R35) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R50-R62)

Documentation Updates:

* Updated `README.md` to reflect the migration from Next.js to Vite and added new package links. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L42-R42) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R53)
* Added new changes to `CHANGES.md` to document the migration and other significant updates.

These changes collectively ensure a smooth transition from Next.js to Vite/React, improve the development workflow, and update the documentation to reflect the latest changes.

### CODE REVIEW

File: src/components/Video/Video.tsx [hunk 1]
```diff
@@ -40,29 +40,45 @@ const VideoElement: React.FC<VideoProps> = (props) => {
...
+      return
+    }
+
+    if (stream) {
       // Chrome で first video frame まで音声が出力されない現象のワークアラウンド
       // 一旦 video tracks を disabled にしておき、 loadedmetadata イベントで有効にする
       // c.f. https://bugs.chromium.org/p/chromium/issues/detail?id=403710
       let originalEnabled: boolean | undefined
-      stream.getVideoTracks().filter((track) => {
+      for (const track of stream.getVideoTracks()) {
```
[the remark this answers is not in the document]
Comment (sile) [reply src/components/Video/Video.tsx:53]: 触った覚えがない箇所なので、どの辺りを見て欲しいのかがすぐには分かりませんでした。

[the remark this answers is not in the document]
Comment (sile) [reply src/components/Video/Video.tsx:53]: 了解です 👌

File: src/components/Video/Video.tsx [hunk 2]
```diff
@@ -40,29 +40,45 @@ const VideoElement: React.FC<VideoProps> = (props) => {
...
       }
 
-      videoRef.current.srcObject = stream
+      videoElement.srcObject = stream
       if (audioOutput && stream.getAudioTracks().length > 0) {
-        videoRef.current.setSinkId(audioOutput)
+        videoElement.setSinkId(audioOutput)
+      }
+
+      return () => {
```
[the remark this answers is not in the document]
Comment (tnamao) [reply src/components/Video/Video.tsx:70]: track.enabled が false になって戻らないケースは、`onloadedmetadata` のイベントが発火する前に、コンポーネントがアンマウントされたケースでしょうか？
実装は問題なさそうですが、何を気にしてクリーンアップしているのかコメントが欲しいと思いました

File: src/components/Video/Video.tsx [hunk 3]
```diff
@@ -40,29 +40,45 @@ const VideoElement: React.FC<VideoProps> = (props) => {
...
+          if (originalEnabled !== undefined) {
+            track.enabled = originalEnabled
+          }
+        }
       }
-    } else if (videoRef.current && stream === null) {
-      videoRef.current.srcObject = null
+    }
+
+    if (stream === null) {
```
Comment (tnamao) [inline src/components/Video/Video.tsx:79]: else 使わないのであれば、48 行目の前に持って行っても良さそうな気がしました

**Review by tnamao — APPROVED** [review]

**Pull request merged**